tests/ui/legacy_numeric_constants.rs

```diff
@@ -0,0 +1,52 @@
+//@run-rustfix
+//@aux-build:proc_macros.rs:proc-macro
+#![allow(clippy::no_effect, deprecated, unused)]
+#![warn(clippy::legacy_numeric_constants)]
+
+#[macro_use]
+extern crate proc_macros;
+
+use std::u128 as _;
+pub mod a {
+    pub use std::u128;
+}
+
+fn main() {
+    std::f32::EPSILON;
+    std::u8::MIN;
+    std::usize::MIN;
+    std::u32::MAX;
+    core::u32::MAX;
+    use std::u32::MAX;
+    MAX;
+    u32::max_value();
+    u8::max_value();
+    u8::min_value();
+    ::std::primitive::u8::MIN;
+    ::std::u8::MIN;
+    ::std::primitive::u8::min_value();
+    std::primitive::u32::max_value();
+    self::a::u128::MAX;
+    // Don't lint
+    use std::f64;
+    f32::EPSILON;
+    u8::MIN;
+    std::f32::consts::E;
+    f64::consts::E;
+    external! {
+        ::std::primitive::u8::MIN;
+        ::std::u8::MIN;
+        ::std::primitive::u8::min_value();
+    }
+}
+
+#[clippy::msrv = "1.42.0"]
+fn msrv_too_low() {
+    // FIXME: Why does this lint??
+    std::u32::MAX;
+}
```

Comment:
This properly also has a similar reason like `span_lint_hir`. The msrv is tracked by  `extract_msrv_attr!(LateContext);` which does it for every node, but you lint once at the end. Either, you have to do the tracking in the visitor or change the logic to lint in the lint pass

Reply:
Yeah it should probably do it like that instead